dagstore.go

```diff
@@ -331,6 +332,59 @@ func (d *DAGStore) AllShardsInfo() AllShardsInfo {
 	return ret
 }
 
+// GC attempts to reclaim the transient files of shards that are currently
+// available but inactive.
+//
+// It is not strictly atomic for now, as it determines which shards to reclaim
+// first, sends operations to the event loop, and waits for them to execute.
+// In the meantime, there could be state transitions that change reclaimability
+// of shards (some shards deemed reclaimable are no longer so, and vice versa).
+//
+// However, the event loop checks for safety prior to deletion, so it will skip
+// over shards that are no longer safe to delete.
+func (d *DAGStore) GC(ctx context.Context) (map[shard.Key]error, error) {
```

Comment:
@raulk Please can we launch a go-routine to do this periodically (configurable period) in the DAG Store ? Otherwise, we'll have to place this burden on all clients.

Reply:
I'm not convinced periodic GC is the best approach, although I filed it here: #56. For now, we probably want to keep it manual, but in the future we will want to monitor disk usage of the transients dir and perform GC when it reaches a watermark, not periodically.